block-producer/src/state_view/mod.rs

```diff
@@ -54,7 +56,22 @@ where
         &self,
         candidate_tx: &ProvenTransaction,
     ) -> Result<(), VerifyTxError> {
-        // 1. soft-check if `tx` violates in-flight requirements.
+        #[cfg(not(test))]
+        // The verification of the transaction proof is disabled in tests for now because
+        // we are using a `DummyProver`, hence proofs generated in tests are invalid.
+        // TODO: Add valid proof generation in tests
+        {
+            // 1. Verify transaction proof
+            //
+            // This check makes sure that the transaction proof that has been attached to the transaction
+            // is valid, errors out if the proof is invalid.
+            let tx_verifier = TransactionVerifier::new(96);
+            tx_verifier
+                .verify(candidate_tx.clone())
+                .map_err(|_| VerifyTxError::InvalidTransactionProof(candidate_tx.id()))?;
```

Comment:
As mentioned in the issue, let's add a config setting - e.g., something like `check_tx_proofs`. For now, it could be hard-coded to `true` and we can expose it via `miden-node.toml` as a separate issue.

Also, let's define a constant for `96` in `miden-base` as discussed in the other comment.

Reply:
I have not clearly understood what you were referring to when saying `a config setting`.

Something like so: 
```rust
/// Block producer specific configuration
#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Serialize, Deserialize)]
pub struct BlockProducerConfig {
    pub endpoint: Endpoint,

    /// Store gRPC endpoint in the format `http://<host>[:<port>]`.
    pub store_url: String,

    pub check_tx_proofs: bool
}
```

Also I don't fully understand when we wouldn't want to not verify proofs ( except in tests ).

Reply:
> Something like so:
> 
> ```rust
> /// Block producer specific configuration
> #[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Serialize, Deserialize)]
> pub struct BlockProducerConfig {
>     pub endpoint: Endpoint,
> 
>     /// Store gRPC endpoint in the format `http://<host>[:<port>]`.
>     pub store_url: String,
> 
>     pub check_tx_proofs: bool
> }
> ```

Yes, this is correct.

> Also I don't fully understand when we wouldn't want to not verify proofs ( except in tests ).

I briefly mentioned this in https://github.com/0xPolygonMiden/miden-node/issues/106#issuecomment-1902598703, but to elaborate more:
- Proofs take non-negligible time to verify (10 - 15ms).
- We already verify proofs in the PRC - so, by the time they get to the block producer they should be already verified.
- Block producer will also verify them later when building a batch (as a part of batch proof generation).

Reply:
Understood, hence the `check_tx_proof` configuration would enable/disable the `verify_tx` proof verification. 

But the batch proof verification would always be done.

Right?